From 229afa6b7c0522796cccf32b0972853b42761326 Mon Sep 17 00:00:00 2001
From: Rikardo-hm <rikard.olander@hostmobility.com>
Date: Thu, 6 Feb 2025 09:07:12 +0000
Subject: [PATCH 1/1]  Use dev board dtsi for hmm with modifications
V.1. Remove unused parts and add i2c buss handling which is the most important part. Disable eth2 that is not supported by hmm board.
Enable otg usb.

V.2 Fixup eeprom, usb cc and led controller i2c address.
    Add gpio leds so led 3 and 4 are lit.
    fix usb-c to operate as host for usb memory flash.

V.3 led 1 is initialized to yellow and turns green when exiting boot.scr (see distro-boot).

v.4 adapt to 2024.04 uboot.

---
 arch/arm/dts/k3-am62-verdin-dev.dtsi          | 113 ++++++------------
 arch/arm/dts/k3-am62-verdin.dtsi              |   5 +
 .../dts/k3-am625-verdin-wifi-dev-u-boot.dtsi  |  84 +++++++++++--
 3 files changed, 113 insertions(+), 89 deletions(-)

diff --git a/arch/arm/dts/k3-am62-verdin-dev.dtsi b/arch/arm/dts/k3-am62-verdin-dev.dtsi
index 6701cb8974b..ff44d0e2a4d 100644
--- a/arch/arm/dts/k3-am62-verdin-dev.dtsi
+++ b/arch/arm/dts/k3-am62-verdin-dev.dtsi
@@ -47,23 +47,13 @@
 /* Verdin ETHs */
 &cpsw3g {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_rgmii1>, <&pinctrl_rgmii2>;
+	pinctrl-0 = <&pinctrl_rgmii1>;
 	status = "okay";
 };
 
 /* MDIO, shared by Verdin ETH_1 (On-module PHY) and Verdin ETH_2_RGMII */
 &cpsw3g_mdio {
 	status = "okay";
-
-	cpsw3g_phy1: ethernet-phy@7 {
-		compatible = "ethernet-phy-ieee802.3-c22";
-		reg = <7>;
-		interrupt-parent = <&main_gpio0>;
-		interrupts = <38 IRQ_TYPE_EDGE_FALLING>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pinctrl_eth2_rgmii_int>;
-		micrel,led-mode = <0>;
-	};
 };
 
 /* Verdin ETH_1 (On-module PHY) */
@@ -71,32 +61,6 @@
 	status = "okay";
 };
 
-/* Verdin ETH_2_RGMII */
-&cpsw_port2 {
-	phy-handle = <&cpsw3g_phy1>;
-	phy-mode = "rgmii-rxid";
-	status = "okay";
-};
-
-/* Verdin PWM_1, PWM_2 */
-&epwm0 {
-	status = "okay";
-};
-
-/* Verdin PWM_3_DSI */
-&epwm1 {
-	status = "okay";
-};
-
-&main_gpio0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_ctrl_sleep_moci>,
-		    <&pinctrl_gpio_5>,
-		    <&pinctrl_gpio_6>,
-		    <&pinctrl_gpio_7>,
-		    <&pinctrl_gpio_8>;
-};
-
 /* Verdin I2C_1 */
 &main_i2c1 {
 	status = "okay";
@@ -111,44 +75,45 @@
 	};
 
 	/* IO Expander */
-	gpio_expander_21: gpio@21 {
-		compatible = "nxp,pcal6416";
-		reg = <0x21>;
-		#gpio-cells = <2>;
+	max7321: gpio-expander@68 {
+		compatible = "maxim,max7320";
+		reg = <0x68>;
 		gpio-controller;
+		#gpio-cells = <2>;
+		gpio-line-names =
+				"green_led_2", "red_led_2",
+				"green_led_1", "red_led_1",
+				"green_led_4", "red_led_4",
+				"green_led_3", "red_led_3";
+		interrupt-controller;
+		#interrupt-cells = <2>;
 	};
 
-	/* Current measurement into module VCC */
-	hwmon@40 {
-		compatible = "ti,ina219";
-		reg = <0x40>;
-		shunt-resistor = <10000>;
-	};
-
-	/* Temperature sensor */
-	sensor@4f {
-		compatible = "ti,tmp75c";
-		reg = <0x4f>;
-	};
-
-	/* EEPROM */
-	eeprom@57 {
-		compatible = "st,24c02", "atmel,24c02";
-		reg = <0x57>;
-		pagesize = <16>;
+	/*USB-C controller*/
+	typec@1d {
+		compatible = "nxp,ptn5150";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_usb1_irq>;
+		interrupt-parent = <&main_gpio0>;
+		interrupts = <38 IRQ_TYPE_LEVEL_HIGH>;
+		reg = <0x1d>;
+		status ="okay";
+ 	};
+
+	eeprom@54 {
+		compatible = "st,24c02", "atmel,24c04";
+		reg = <0x54>;
+		status = "okay";
+		//vcc-supply = <&reg_base_3v3>;
+ 	};
+	/* Add HSM module */
+	hsm@48 {
+		compatible = "There is no driver";
+		reg = <0x48>;
+		status = "okay";
 	};
 };
 
-/* Verdin I2C_2_DSI */
-&main_i2c2 {
-	status = "okay";
-};
-
-/* Verdin I2C_4_CSI */
-&main_i2c3 {
-	status = "okay";
-};
-
 /* Verdin CAN_1 */
 &main_mcan0 {
 	status = "okay";
@@ -199,11 +164,6 @@
 	status = "okay";
 };
 
-/* Verdin QSPI_1 */
-&ospi0 {
-	status = "okay";
-};
-
 /* Verdin SD_1 */
 &sdhci1 {
 	ti,driver-strength-ohm = <33>;
@@ -217,15 +177,16 @@
 
 &usb0 {
 	status = "okay";
+	dr_mode = "host";
 };
 
 /* Verdin USB_2 */
 &usbss1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &usb1 {
-	status = "okay";
+	status = "disabled";
 };
 
 /* Verdin CTRL_WAKE1_MICO# */
diff --git a/arch/arm/dts/k3-am62-verdin.dtsi b/arch/arm/dts/k3-am62-verdin.dtsi
index d4f8776c927..bc79501927e 100644
--- a/arch/arm/dts/k3-am62-verdin.dtsi
+++ b/arch/arm/dts/k3-am62-verdin.dtsi
@@ -688,6 +688,11 @@
 			AM62X_IOPAD(0x0060, PIN_OUTPUT, 1) /*  (R25) GPMC0_AD9.VOUT0_DATA17 */
 		>;
 	};
+	pinctrl_usb1_irq: usb1-irq-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x009c, PIN_INPUT, 7) /* (V25) GPMC0_WAIT1.GPIO0_38 */ /* GPIO_SODIMM_189 */
+		>;
+	};
 };
 
 &mcu_pmx0 {
diff --git a/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi b/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi
index 206df610461..965c6d571b6 100644
--- a/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi
+++ b/arch/arm/dts/k3-am625-verdin-wifi-dev-u-boot.dtsi
@@ -8,7 +8,7 @@
 / {
 	aliases {
 		eeprom0 = &eeprom_module;
-		eeprom1 = &eeprom_carrier_board;
+		eeprom1 = &eeprom_hmm_board;
 		eeprom2 = &eeprom_display_adapter;
 	};
 
@@ -19,6 +19,71 @@
 	memory@80000000 {
 		bootph-all;
 	};
+	/*	The LEDs are read from the highest to lowest position on the module when the PCB is in horizontal position.
+		Default: LED 1 is yellow (green+red) and the rest are off.
+	*/
+	leds {
+		compatible = "gpio-leds";
+
+		green_led_1 {
+			label = "green:led_1";
+			function = "led_1";
+			gpios = <&max7321 2 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		red_led_1 {
+			label = "red:led_1";
+			function = "led_1";
+			gpios = <&max7321 3 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		green_led_2 {
+			label = "green:led_2";
+			function = "led_2";
+			gpios = <&max7321 0 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		red_led_2 {
+			label = "red:led_2";
+			function = "led_2";
+			gpios = <&max7321 1 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_3 {
+			label = "green:led_3";
+			function = "led_3";
+			gpios = <&max7321 6 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		red_led_3 {
+			label = "red:led_3";
+			function = "led_3";
+			gpios = <&max7321 7 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		green_led_4 {
+			label = "green:led_4";
+			function = "led_4";
+			gpios = <&max7321 4 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+		red_led_4 {
+			label = "red:led_4";
+			function = "led_4";
+			gpios = <&max7321 5 GPIO_ACTIVE_LOW>;
+			default-state = "off";
+		};
+
+
+	};
+
 
 	sysinfo {
 		compatible = "toradex,sysinfo";
@@ -82,10 +147,6 @@
 	bootph-all;
 };
 
-&cpsw3g_phy1 {
-	bootph-all;
-};
-
 &cpsw_port1 {
 	bootph-all;
 };
@@ -129,10 +190,9 @@
 	};
 
 	/* EEPROM on carrier board */
-	eeprom_carrier_board: eeprom@57 {
+	eeprom_hmm_board: eeprom@54 {
 		compatible = "i2c-eeprom";
-		reg = <0x57>;
-		pagesize = <16>;
+		reg = <0x54>;
 	};
 };
 
@@ -183,14 +243,12 @@
 };
 
 &usb0 {
-	dr_mode = "peripheral";
-	/* Since role switching is not supported in U-Boot */
-	/delete-property/ extcon;
-	bootph-pre-ram;
+	dr_mode = "host";
+	status = "okay";
 };
 
 &usbss0 {
-	bootph-pre-ram;
+	status = "okay";
 };
 
 &verdin_ctrl_sleep_moci {
-- 
2.25.1

